auth0/v3/management/rest.py

```diff
@@ -1,16 +1,87 @@
-import sys
-import json
 import base64
+import json
+import sys
+import time
+
 import requests
+
 from ..exceptions import Auth0Error
 
 
-class RestClient(object):
+class NopRateLimiter(object):
+    def update(self, headers):
+        pass
+
+    def ensure_limit(self):
+        pass
+
+
+class RateLimiter(object):
+    def __init__(self):
+        self.ratelimit_reset = int(time.time()) - 1
+        self.ratelimit_remaining = 1
+
+    def update(self, headers):
+        if 'x-ratelimit-remaining' in headers:
+            self.ratelimit_remaining = int(headers['x-ratelimit-remaining'])
+
+        if 'x-ratelimit-reset' in headers:
+            self.ratelimit_reset = int(headers['x-ratelimit-reset'])
+
+    def ensure_limit(self):
+        self.ratelimit_remaining -= 1
+
+        if self.ratelimit_remaining < 0:
+            sleep_time = self.ratelimit_reset - int(time.time())
+            if sleep_time > 0:
+                time.sleep(sleep_time)
+
+
+def extract_content(func):
```

Comment:
shouldn't this be called `process_response` since AFAIK it's doing the same as before?

Reply:
Yep, will rename this.